cmd/certsuite/upload/results_spreadsheet/results_spreadsheet.go

```diff
@@ -0,0 +1,367 @@
+		log.Fatalf("Failed to mark results file path as required parameter: %v", err)
+		return nil
+	}
+
+	err = uploadResultSpreadSheetCmd.MarkFlagRequired("dest-url")
+	if err != nil {
+		log.Fatalf("Failed to mark dest url path as required parameter: %v", err)
+		return nil
+	}
+
+	return uploadResultSpreadSheetCmd
+}
+
+func readCSV(fp string) ([][]string, error) {
+	file, err := os.Open(fp)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	return records, nil
+}
+
+func createSheetsAndDriveServices() (sheetService *sheets.Service, driveService *drive.Service, err error) {
+	ctx := context.Background()
+	b, err := os.ReadFile("cmd/certsuite/upload/results_spreadsheet/credentials.json")
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to read client secret file: %v", err)
+	}
+
+	// If modifying these scopes, delete your previously saved token.json.
+	config, err := google.ConfigFromJSON(b, sheets.SpreadsheetsScope, drive.DriveScope)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to parse client secret file to config: %v", err)
+	}
+	client, err := getClient(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to get client: %v", err)
+	}
+
+	sheetSrv, err := sheets.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to retrieve Sheets client: %v", err)
+	}
+
+	driveSrv, err := drive.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to retrieve Drive client: %v", err)
+	}
+
+	return sheetSrv, driveSrv, nil
+}
+
+func prepareRecordsForSpreadSheet(records [][]string) []*sheets.RowData {
+	var rows []*sheets.RowData
+	for _, row := range records {
+		var rowData []*sheets.CellData
+		for _, col := range row {
+			var val string
+			// cell content cannot exceed 50,000 letters.
+			if len(col) > cellContantLimit {
+				col = col[:cellContantLimit]
+			}
+			// use space for empty values to avoid cells overlapping
+			if col == "" {
+				val = " "
+			}
+			// avoid line breaks in cell
+			val = strings.ReplaceAll(strings.ReplaceAll(col, "\r\n", " "), "\n", " ")
+
+			rowData = append(rowData, &sheets.CellData{
+				UserEnteredValue: &sheets.ExtendedValue{StringValue: &val},
+			})
+		}
+		rows = append(rows, &sheets.RowData{Values: rowData})
+	}
+	return rows
+}
+
+func createSingleWorkloadRawResultsSheet(rawResultsSheet *sheets.Sheet, workloadName string) (*sheets.Sheet, error) {
+	// Initialize sheet with headers
+	filteredRows := []*sheets.RowData{{Values: []*sheets.CellData{
+		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Owner/TechLead Conclusion")}},
+		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Next Step Actions")}},
+	}}}
+	filteredRows[0].Values = append(filteredRows[0].Values, rawResultsSheet.Data[0].RowData[0].Values...)
```

Comment:
in line 142 I added a comment saying "Initialize sheet with headers", afterwards in line 156 I'm explaining the filter - "add to sheet only rows of given workload name".
But I get why it might be a bit confusing.. maybe I can initialize the headers in a different var and than add it to the `filteredRows`. What do you think?

Reply:
No, it's ok as it is now, and I finally understood it. I'd rather be a bit more verbose on the purpose of certain code blocks, like this one (lines 143-147). Also, I believe this functions should be properly docummented. If I'm not wrong, something like this would be fine:
```
// createSingleWorkloadRawResultsSheet creates a new spreadsheet with test case results of a single workload,
// extracted from rawResultsSheets (which may contain the results of several workloads).
// The spreadsheet will use the same header columns as the rawResultsSheet, but will also add two extra columns:
// - "Owner/TechLead Conclusion": the partner/user is expected to add the name of the workload owner that should lead the fix
//   of this test case result.
// - "Next Step Actions": the partner/user may use this column to add the follow-up actions to fix this test case result.
func createSingleWorkloadRawResultsSheet(rawResultsSheet *sheets.Sheet, workloadName string) (*sheets.Sheet, error) {
	// Initialize sheet with the two new column headers only.
	filteredRows := []*sheets.RowData{{Values: []*sheets.CellData{
		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Owner/TechLead Conclusion")}},
		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Next Step Actions")}},
	}}}

	// Add existing column headers from the rawResultsSheet
	filteredRows[0].Values = append(filteredRows[0].Values, rawResultsSheet.Data[0].RowData[0].Values...)
	...
```